app/src/main/java/com/radixdlt/android/helper/CustomTabsHelper.kt

```diff
@@ -0,0 +1,140 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Modifications copyright (C) 2019 Radix DLT
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.radixdlt.android.helper
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.net.Uri
+import android.text.TextUtils
+import timber.log.Timber
+import java.util.ArrayList
+
+/**
+ * Helper class for Custom Tabs.
+ */
+object CustomTabsHelper {
+    private const val STABLE_PACKAGE = "com.android.chrome"
+    private const val BETA_PACKAGE = "com.chrome.beta"
+    private const val DEV_PACKAGE = "com.chrome.dev"
+    private const val LOCAL_PACKAGE = "com.google.android.apps.chrome"
+    private const val ACTION_CUSTOM_TABS_CONNECTION =
+        "android.support.customtabs.action.CustomTabsService"
+
+    private var sPackageNameToUse: String? = null
+
+    /**
+     * Goes through all apps that handle VIEW intents and have a warmup service. Picks
+     * the one chosen by the user if there is one, otherwise makes a best effort to return a
+     * valid package name.
+     *
+     * This is **not** threadsafe.
+     *
+     * @param context [Context] to use for accessing [PackageManager].
+     * @return The package name recommended to use for connecting to custom tabs related components.
+     */
+    fun getPackageNameToUse(context: Context): String? {
+        if (sPackageNameToUse != null) return sPackageNameToUse
+
+        val pm = context.packageManager
+        // Get default VIEW intent handler.
+        val activityIntent = Intent(
+            Intent.ACTION_VIEW, Uri.parse("http://www.example.com")
```

Comment:
This `Uri` seems funky. Could you please verify it's by intention?

Reply:
Need to check this out

Reply:
For this, I don't like to make assumptions but I believe it simply needs a valid web uri. It's a helper class written by the Chrome Custom Tab guys and it works so... It should be fine!